src/puffin/src/puffin_manager/stager/bounded_stager.rs

```diff
@@ -337,6 +368,9 @@ impl BoundedStager {
     }
 
     /// Recovers the staging area by iterating through the staging directory.
+    ///
+    /// Note: It can't recover the mapping between puffin files and keys, so TTL
+    ///       is configured to purge the dangling files and directories.
```

Comment:
non-blocking: Now we hash the file name and key to get the cache key so we can't recover the relationship, we may consider a better way to compute the cache key.

Reply:
It seems that there is no solution to reversibly encode the concatenation of two strings without introducing any assumptions?

Reply:
Yes, we have to use `/` as delimiter and use the key as path directly. The drawback of encoding the key is that we don't know which puffin file the cache entry belongs to.